frontend/src/__tests__/cypress/cypress/e2e/projects/projectDetails.cy.ts

```diff
@@ -0,0 +1,218 @@
+    mockProjectK8sResource({}),
+  );
+
+  cy.intercept(
+    {
+      pathname: `/api/k8s/apis/route.openshift.io/v1/namespaces/test-project/routes/ds-pipeline-pipelines-definition`,
+    },
+    mockRouteK8sResource({
+      notebookName: 'ds-pipeline-pipelines-definition',
+    }),
+  );
+
+  cy.intercept(
+    { pathname: '/api/k8s/api/v1/namespaces/test-project/persistentvolumeclaims' },
+    mockK8sResourceList(isEmpty ? [] : [mockPVCK8sResource({})]),
+  );
+
+  cy.intercept(
+    {
+      pathname: '/api/k8s/apis/image.openshift.io/v1/namespaces/opendatahub/imagestreams',
+    },
+    mockK8sResourceList(
+      isEmpty || isUnknown
+        ? []
+        : [
+            mockImageStreamK8sResource({
+              name: imageStreamName,
+              displayName: 'Test image',
+              opts: {
+                metadata: {
+                  labels: {
+                    'opendatahub.io/notebook-image': isEnabled,
+                  },
+                },
+                spec: {
+                  tags: [
+                    {
+                      name: 'latest',
+                    },
+                  ],
+                },
+                status: {
+                  tags: [
+                    {
+                      tag: 'latest',
+                    },
+                  ],
+                },
+              },
+            }),
+          ],
+    ),
+  );
+
+  cy.intercept(
+    { pathname: '/api/k8s/apis/kubeflow.org/v1/namespaces/test-project/notebooks' },
+    mockK8sResourceList(
+      isEmpty
+        ? []
+        : [
+            mockNotebookK8sResource({
+              opts: {
+                spec: {
+                  template: {
+                    spec: {
+                      containers: [
+                        {
+                          name: 'test-notebook',
+                          image: 'test-image:latest',
+                        },
+                      ],
+                    },
+                  },
+                },
+                metadata: {
+                  name: 'test-notebook',
+                  labels: {
+                    'opendatahub.io/notebook-image': 'true',
+                  },
+                  annotations: {
+                    'opendatahub.io/image-display-name': isUnknown ? '' : 'Test image',
+                  },
+                },
+              },
+            }),
+          ],
+    ),
+  );
+
+  cy.intercept(
+    {
+      pathname: '/api/k8s/apis/route.openshift.io/v1/namespaces/test-project/routes/test-notebook',
+    },
+    mockRouteK8sResource({ notebookName: 'test-notebook' }),
+  );
+
+  cy.intercept(
+    {
+      method: 'POST',
+      pathname: '/api/proxy/apis/v1beta1/pipelines',
+    },
+    buildMockPipelines(isEmpty ? [] : [mockPipelineKF({})]),
+  );
+};
+
+describe('Project Details', () => {
+  it('Empty project', () => {
+    initIntercepts({ isEmpty: true });
```

Comment:
can you put these in `beforeEach()`? both `initIntercepts` and `projectDetails.visit('test-project');`

Reply:
initIntercepts has different parameters passed in every test cases

Reply:
oh yes, didn't see that, my bad.